### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by egichuhis**
Hello, please review this PR.

Required:

- use two decorators in order to capitalize and trim people's names.

1. Create a class Nameable. Implement a method called correct_name that will raise a NotImplementedError.
2. Turn Person class to Nameable, has a method correct_name implemented. It should simply return the name attribute.
3. Add base Decorator, Implement the correct_name method that returns the result of the correct_name method of the @nameable.
4. Prepare CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -19,3 +33,29 @@ def of_age?
...
+    super().capitalize
+  end
+end
+
+# TrimmerDecorator
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    super()[0..9]
+  end
+end
```
Comment (algerina) [inline person.rb]: - [ ] When trying the testing your code as per the requirements I had the following errors, Kindly try to fix your classes:



Screenshot 2023-11-09 at 2 33 46 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @egichuhis ,

Great job so far! However you have some minor issues to work on.

May I highlight the following: 
-  Well done working on the different classes👍 ✔
- Your code is well organized✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: person.rb (hunk 1 above)
Comment (egichuhis) [reply person.rb]: Thanks @algerina for noting that. Code now runs without a hitch.

**Comment by egichuhis** [comment]
In your IRB session, run the require_relative 'person' first before running the checks

`irb(main):001:0> require_relative 'person'`

irb

**Review by mckent05 — APPROVED** [review]
Hi @egichuhis, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job implementing decorator classes 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**